### ISSUE TITLE
bug: minikube setup

### ISSUE THREAD

**Issue by bahtou** [apache/apisix-ingress-controller#338]
### Issue description

Trying to go through the minikube setup here:
https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/deployments/minikube.md

The installation works fine. It is when I try to test the connection by following the practice example that I run into issues:
https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md#test

After applying the `httpbin-route.yaml` I expected the route to be registered in the Apisix so after running `kubectl exec -it -n ${namespace of Apache APISIX} ${Pod name of Apache APISIX} -- curl http://127.0.0.1:9180/apisix/admin/routes -H 'X-API-Key: edd1c9f034335f136f87ad84b625c8f1'` the returned payload shows no record of the httpbin.

And running `kubectl exec -it -n ${namespace of Apache APISIX} ${Pod name of Apache APISIX} -- curl http://127.0.0.1:9080/headers -H 'Host: local.httpbin.org'` returns a 404.

Are the docs missing something? 

**UPDATE**:
Ran through the process from the beginning:

```
helm install apisix apisix/apisix \
  --set allow.ipList="{10.244.0.0/16}" \
  --namespace ingress-apisix
kubectl get service --namespace ingress-apisix
```

```
helm install apisix-ingress-controller apisix/apisix-ingress-controller \
  --set image.tag=0.4.0 \
  --set config.apisix.baseURL=http://apisix-admin:9180/apisix/admin \
  --set config.apisix.adminKey=edd1c9f034335f136f87ad84b625c8f1 \
  --namespace ingress-apisix
```

Now running the following returns a `403 Forbidden`:
`kk exec -it -n ingress-apisix apisix-646684ff68-7st6l -- curl http://127.0.0.1:9180/apisix/admin/routes -H 'X-API-Key: edd1c9f034335f136f87ad84b625c8f1'`

I am using the same API Key so not sure why this is erring.

**Comment by tokers** [comment]
Basically, the `403` forbidden is returned due to the remote address is not in the allow list. As per your description, you try to access the Admin API from localhost but you just allow IPs from `10.244.0.0/16`.

As for the `httpbin-route`, actually it still uses the ApisixRoute/v1 while we recommend to use ApisixRoute/v2alpha1, the corresponding version of `httpbin-route` in v2alpha1 is just below:

```yaml
# httpbin-route.yaml
apiVersion: apisix.apache.org/v2alpha1
kind: ApisixRoute
metadata:
  name: httpserver-route
spec:
  http:
  - name: rule1
    match:
      hosts:
      - local.httpbin.org
      paths:
      - /*
      backends:
      - serviceName: httpbin
        servicePort: 80
```

**Comment by tokers** [comment]
@bahtou By the way, after you run this case successfully, would you like to submit a PR to update the `httpbin-route` in https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md#test? Thanks!

**Comment by bahtou** [comment]
@tokers thanks for the assist. When apply the `httpbin-route.yaml` I get the following: `no matches for kind "ApisixRoute" in version "apisix.apache.org/v2alpha1"`

**Comment by tokers** [comment]
> @tokers thanks for the assist. When applying the `httpbin-route.yaml` I get the following: `no matches for kind "ApisixRoute" in version "apisix.apache.org/v2alpha1"`

Oops, what's the version of your apisix-ingress-controller, you may try to install the latest one.

**Comment by bahtou** [comment]
@tokers Uisng this one: https://github.com/apache/apisix-ingress-controller/releases/tag/0.4.0
  --set image.tag=0.4.0 \

**Comment by tokers** [comment]
@bahtou You may try to install apisix-ingress-controller from the helm repo: https://charts.apiseven.com

**Comment by bahtou** [comment]
> helm install apisix-ingress-controller apisix/apisix-ingress-controller \
  --set image.tag=0.4.0 \
  --set config.apisix.baseURL=http://apisix-admin:9180/apisix/admin \
  --set config.apisix.adminKey=edd1c9f034335f136f87ad84b625c8f1 \
  --namespace ingress-apisix

Does this not install from the helm repo? I added the repo the step before that one: https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/deployments/minikube.md#install-apisix

**Comment by tokers** [comment]
> > helm install apisix-ingress-controller apisix/apisix-ingress-controller 
> >   --set image.tag=0.4.0 
> >   --set config.apisix.baseURL=http://apisix-admin:9180/apisix/admin 
> >   --set config.apisix.adminKey=edd1c9f034335f136f87ad84b625c8f1 
> >   --namespace ingress-apisix
> 
> Does this not install from the helm repo? I added the repo the step before that one: https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/deployments/minikube.md#install-apisix

Have you updated the repo?

**Comment by bahtou** [comment]
yes. This is the repo charts
>NAME                            	CHART VERSION	APP VERSION	DESCRIPTION
apisix/apisix                   	0.2.0        	2.3.0      	A Helm chart for Apache APISIX
apisix/apisix-dashboard         	0.1.1        	2.3.0      	A Helm chart for Apache APISIX Dashboard
apisix/apisix-ingress-controller	0.3.0        	0.4.0      	Apache APISIX Ingress Controller for Kubernetes

**Comment by tokers** [comment]
It's weird, Is the `ApisixRoute/v2alpha1` in the CRD list?

```shell
k get crds | grep "apisixroute"
```

**Comment by bahtou** [comment]
`apisixroutes.apisix.apache.org      2021-04-06T01:43:18Z`

**Comment by tokers** [comment]
> `apisixroutes.apisix.apache.org 2021-04-06T01:43:18Z`
> 
> Kubernetes version:
> 
> > Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-21T20:21:49Z", GoVersion:"go1.15.8", Compiler:"gc", Platform:"darwin/amd64"}
> > Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.2", GitCommit:"f5743093fd1c663cb0cbc89748f730662345d44d", GitTreeState:"clean", BuildDate:"2020-09-16T13:32:58Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}

I'm afraid that i didn't get useful information there 😂, would you like to just paste all output from executing `k get crds`.

**Comment by bahtou** [comment]
:)
> $ kubectl get crds
NAME                                CREATED AT
apisixroutes.apisix.apache.org      2021-04-06T01:43:18Z
apisixtlses.apisix.apache.org       2021-04-06T01:43:18Z
apisixupstreams.apisix.apache.org   2021-04-06T01:43:18Z

**Comment by tokers** [comment]
@bahtou It seems the newest CRDs were not installed correctly. Have you tried to remove and install it? I doubt that CRDs are not configured when you installing the newest apisix-ingress-controller.

**Comment by bahtou** [comment]
@tokers I started from scratch by blasting minikube.
I went through the first two instructions here: https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/deployments/minikube.md#install-apisix
1. install apisix
2. install apisix-ingress-controller

>$ kubectl api-resource | grep 'ApisixRoute'
NAME                              SHORTNAMES   APIVERSION                        NAMESPACED   KIND
apisixroutes                      ar           apisix.apache.org/v1              true         ApisixRoute

The APIVERSION is not `ApisixRoute/v2alpha1`. And the error is `no matches for kind "ApisixRoute" in version "apisix.apache.org/v2alpha1"`

**Comment by tokers** [comment]
Well, you may try to apply the newest crds in samples/deploy. In fact, the newest charts in the https://charts.apiseven.com already fixed this issue.

**Comment by bahtou** [comment]
@tokers have you gone through the instruction set? Is it working for you with a fresh minikube cluster? All methods you have mentioned have not worked.

**Comment by bahtou** [comment]
@tokers I created the CRDs using the `samples/deploy` directory
This worked and the `httpbin-route.yaml` was created successfully with `apisix.apache.org/v2alpha1`

Looking at the pod `apisix-ingress-controller-847df99d4c-v5g2p`, initially it is up and running fine. But then it crashes and the following are the logs:
```
2021-04-09T00:41:08+08:00 info ingress/ingress.go:113 use configuration
{
"log_level": "info",
"log_output": "stderr",
"http_listen": ":8080",
"enable_profiling": true,
"kubernetes": {
"kubeconfig": "",
"resync_interval": "1m0s",
"app_namespaces": [
""
],
"election_id": "ingress-apisix-leader",
"ingress_class": "apisix",
"ingress_version": "networking/v1",
"apisix_route_version": "apisix.apache.org/v2alpha1"
},
"apisix": {
"base_url": "http://apisix-admin:9180/apisix/admin",
"admin_key": "edd1c9f034335f136f87ad84b625c8f1"
}
}
I0409 00:41:08.128038 1 leaderelection.go:243] attempting to acquire leader lease ingress-apisix/ingress-apisix-leader...
2021-04-09T00:41:08+08:00 info controller/controller.go:198 LeaderElection {"message": "apisix-ingress-controller-847df99d4c-v5g2p became leader", "event_type": "Normal"}
I0409 00:41:08.146784 1 leaderelection.go:253] successfully acquired lease ingress-apisix/ingress-apisix-leader
2021-04-09T00:41:08+08:00 info controller/controller.go:273 controller now is running as leader {"namespace": "ingress-apisix", "pod": "apisix-ingress-controller-847df99d4c-v5g2p"}
2021-04-09T00:41:08+08:00 warn apisix/cluster.go:235 waiting cluster to ready, it may takes a while
2021-04-09T00:41:08+08:00 info apisix/cluster.go:113 syncing cache {"cluster": ""}
2021-04-09T00:41:08+08:00 warn controller/controller.go:236 found a new leader apisix-ingress-controller-847df99d4c-v5g2p
2021-04-09T00:41:08+08:00 info apisix/cluster.go:120 cache synced {"cost_time": "14.051871ms", "cluster": ""}
2021-04-09T00:41:08+08:00 warn apisix/cluster.go:241 cluster now is ready, cost time 14.277115ms
2021-04-09T00:41:08+08:00 info controller/endpoint.go:64 endpoints controller started
2021-04-09T00:41:08+08:00 info controller/apisix_upstream.go:56 ApisixUpstream controller started
2021-04-09T00:41:08+08:00 info controller/ingress.go:59 ingress controller started
2021-04-09T00:41:08+08:00 info controller/apisix_route.go:57 ApisixRoute controller started
2021-04-09T00:41:08+08:00 info controller/secret.go:62 secret controller started
2021-04-09T00:41:08+08:00 warn apisix/upstream.go:113 upstream not found {"fullname": "kube-system_metrics-server_443", "url": "http://apisix-admin:9180/apisix/admin/upstreams/4a344633", "cluster": ""}
2021-04-09T00:41:08+08:00 warn controller/endpoint.go:140 upstream is not referenced {"cluster": "name=; base_url=http://apisix-admin:9180/apisix/admin", "upstream": "kube-system_metrics-server_443"}
2021-04-09T00:41:08+08:00 warn apisix/upstream.go:113 upstream not found {"fullname": "ingress-apisix_apisix-etcd-headless_2380", "url": "http://apisix-admin:9180/apisix/admin/upstreams/82c8442b", "cluster": ""}
2021-04-09T00:41:08+08:00 warn controller/endpoint.go:140 upstream is not referenced {"cluster": "name=; base_url=http://apisix-admin:9180/apisix/admin", "upstream": "ingress-apisix_apisix-etcd-headless_2380"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x15d2e7d]

goroutine 177 [running]:
github.com/apache/apisix-ingress-controller/pkg/kube/translation.(*translator).TranslateRouteV2alpha1(0xc0000c3110, 0xc0003bb680, 0x7, 0xc00060dee8, 0x10, 0x1c95760, 0xc00032f640, 0x0, 0x0, 0x0)
/build/controller/pkg/kube/translation/apisix_route.go:110 +0x74d
github.com/apache/apisix-ingress-controller/pkg/ingress/controller.(*apisixRouteController).sync(0xc00032e000, 0x1c95560, 0xc00053b080, 0xc00063fe00, 0x1a70df8, 0x0)
/build/controller/pkg/ingress/controller/apisix_route.go:142 +0x39e
github.com/apache/apisix-ingress-controller/pkg/ingress/controller.(*apisixRouteController).runWorker(0xc00032e000, 0x1c95560, 0xc00053b080)
/build/controller/pkg/ingress/controller/apisix_route.go:78 +0x57
created by github.com/apache/apisix-ingress-controller/pkg/ingress/controller.(*apisixRouteController).run
/build/controller/pkg/ingress/controller/apisix_route.go:66 +0x1ef
```
The tests fail (https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md#test) and I assume the pod failure is the issue

**Comment by tokers** [comment]
> @tokers I created the CRDs using the `samples/deploy` directory
> This worked and the `httpbin-route.yaml` was created successfully with `apisix.apache.org/v2alpha1`
> 
> Looking at the pod `apisix-ingress-controller-847df99d4c-v5g2p`, initially it is up and running fine. But then it crashes and the following are the logs:
> 
> ```
> 2021-04-09T00:41:08+08:00 info ingress/ingress.go:113 use configuration
> {
> "log_level": "info",
> "log_output": "stderr",
> "http_listen": ":8080",
> "enable_profiling": true,
> "kubernetes": {
> "kubeconfig": "",
> "resync_interval": "1m0s",
> "app_namespaces": [
> ""
> ],
> "election_id": "ingress-apisix-leader",
> "ingress_class": "apisix",
> "ingress_version": "networking/v1",
> "apisix_route_version": "apisix.apache.org/v2alpha1"
> },
> "apisix": {
> "base_url": "http://apisix-admin:9180/apisix/admin",
> "admin_key": "edd1c9f034335f136f87ad84b625c8f1"
> }
> }
> I0409 00:41:08.128038 1 leaderelection.go:243] attempting to acquire leader lease ingress-apisix/ingress-apisix-leader...
> 2021-04-09T00:41:08+08:00 info controller/controller.go:198 LeaderElection {"message": "apisix-ingress-controller-847df99d4c-v5g2p became leader", "event_type": "Normal"}
> I0409 00:41:08.146784 1 leaderelection.go:253] successfully acquired lease ingress-apisix/ingress-apisix-leader
> 2021-04-09T00:41:08+08:00 info controller/controller.go:273 controller now is running as leader {"namespace": "ingress-apisix", "pod": "apisix-ingress-controller-847df99d4c-v5g2p"}
> 2021-04-09T00:41:08+08:00 warn apisix/cluster.go:235 waiting cluster to ready, it may takes a while
> 2021-04-09T00:41:08+08:00 info apisix/cluster.go:113 syncing cache {"cluster": ""}
> 2021-04-09T00:41:08+08:00 warn controller/controller.go:236 found a new leader apisix-ingress-controller-847df99d4c-v5g2p
> 2021-04-09T00:41:08+08:00 info apisix/cluster.go:120 cache synced {"cost_time": "14.051871ms", "cluster": ""}
> 2021-04-09T00:41:08+08:00 warn apisix/cluster.go:241 cluster now is ready, cost time 14.277115ms
> 2021-04-09T00:41:08+08:00 info controller/endpoint.go:64 endpoints controller started
> 2021-04-09T00:41:08+08:00 info controller/apisix_upstream.go:56 ApisixUpstream controller started
> 2021-04-09T00:41:08+08:00 info controller/ingress.go:59 ingress controller started
> 2021-04-09T00:41:08+08:00 info controller/apisix_route.go:57 ApisixRoute controller started
> 2021-04-09T00:41:08+08:00 info controller/secret.go:62 secret controller started
> 2021-04-09T00:41:08+08:00 warn apisix/upstream.go:113 upstream not found {"fullname": "kube-system_metrics-server_443", "url": "http://apisix-admin:9180/apisix/admin/upstreams/4a344633", "cluster": ""}
> 2021-04-09T00:41:08+08:00 warn controller/endpoint.go:140 upstream is not referenced {"cluster": "name=; base_url=http://apisix-admin:9180/apisix/admin", "upstream": "kube-system_metrics-server_443"}
> 2021-04-09T00:41:08+08:00 warn apisix/upstream.go:113 upstream not found {"fullname": "ingress-apisix_apisix-etcd-headless_2380", "url": "http://apisix-admin:9180/apisix/admin/upstreams/82c8442b", "cluster": ""}
> 2021-04-09T00:41:08+08:00 warn controller/endpoint.go:140 upstream is not referenced {"cluster": "name=; base_url=http://apisix-admin:9180/apisix/admin", "upstream": "ingress-apisix_apisix-etcd-headless_2380"}
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x15d2e7d]
> 
> goroutine 177 [running]:
> github.com/apache/apisix-ingress-controller/pkg/kube/translation.(*translator).TranslateRouteV2alpha1(0xc0000c3110, 0xc0003bb680, 0x7, 0xc00060dee8, 0x10, 0x1c95760, 0xc00032f640, 0x0, 0x0, 0x0)
> /build/controller/pkg/kube/translation/apisix_route.go:110 +0x74d
> github.com/apache/apisix-ingress-controller/pkg/ingress/controller.(*apisixRouteController).sync(0xc00032e000, 0x1c95560, 0xc00053b080, 0xc00063fe00, 0x1a70df8, 0x0)
> /build/controller/pkg/ingress/controller/apisix_route.go:142 +0x39e
> github.com/apache/apisix-ingress-controller/pkg/ingress/controller.(*apisixRouteController).runWorker(0xc00032e000, 0x1c95560, 0xc00053b080)
> /build/controller/pkg/ingress/controller/apisix_route.go:78 +0x57
> created by github.com/apache/apisix-ingress-controller/pkg/ingress/controller.(*apisixRouteController).run
> /build/controller/pkg/ingress/controller/apisix_route.go:66 +0x1ef
> ```
> 
> The tests fail (https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md#test) and I assume the pod failure is the issue

I'm sorry because in `0.4.0` we don't have a field called `backends` but only `backend`, so you can use the following snippet instead:

```yaml
# httpbin-route.yaml
apiVersion: apisix.apache.org/v2alpha1
kind: ApisixRoute
metadata:
  name: httpserver-route
spec:
  http:
  - name: rule1
    match:
      hosts:
      - local.httpbin.org
      paths:
      - /*
      backends:
      - serviceName: httpbin
        servicePort: 80
```

You may try to use the apisix-ingress-controller `0.5.0` .

**Comment by bahtou** [comment]
@tokers sounds good. Happy to test when it is released. Thanks!

**Comment by tokers** [comment]
@bahtou We have released 0.5.0 version, would you like to try it?

**Comment by bahtou** [comment]
@tokers 
Following the documentation here: https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/deployments/minikube.md#install-apisix

```
helm install apisix-ingress-controller apisix/apisix-ingress-controller \
  --set image.tag=0.5.0 \
  --set config.apisix.baseURL=http://apisix-admin:9180/apisix/admin \
  --set config.apisix.adminKey=edd1c9f034335f136f87ad84b625c8f1 \
  --namespace ingress-apisix
```
`error: unable to recognize "httpbin-route.yaml": no matches for kind "ApisixRoute" in version "apisix.apache.org/v2alpha1"`

yaml file
```
apiVersion: apisix.apache.org/v2alpha1
kind: ApisixRoute
metadata:
  name: httpserver-route
spec:
  http:
  - name: rule1
    match:
      hosts:
      - local.httpbin.org
      paths:
      - /*
      backend:
      - serviceName: httpbin
        servicePort: 80
```

I also installed the crds via the `samples/deploy/crd/v1beta1` but then get this error:
```The ApisixRoute "httpserver-route" is invalid:
* <nil>: Invalid value: "": "spec.http" must validate one and only one schema (oneOf). Found none valid
* spec.http.backend: Required value```

**Comment by tokers** [comment]
@bahtou The indent of `backend` is wrong, it should be aligned with `match`, not inside `match`.

**Comment by fhuzero** [comment]
I have encountered the same problem when following [proxy-the-httpbin-service](https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md#resource-delivery) practice. It seems the issue here is still unresolved. After reading [ApisixRoute CRD](https://github.com/apache/apisix-ingress-controller/blob/master/samples/deploy/crd/v1beta1/ApisixRoute.yaml), I have figured out a correct format to apply ApisixRoute by `kubectl apply -f httpbin-route.yaml`. I have initiated Pull Request #546 to fix it.

**Comment by fhuzero** [comment]
After successfully applied `httpbin-route.yaml`, I continued follow instructions [here](https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md#test) to check whether the resource was delivered to an Apache APISIX Pod. After executing command
```
kubectl exec -it -n ingress-apisix apisix-6fc55984cb-hwvjq -- curl http://127.0.0.1:9180/apisix/admin/routes -H 'X-API-Key: edd1c9f034335f136f87ad84b625c8f1'
```
I got output:
```
Defaulted container "apisix" out of: apisix, wait-etcd (init)
{"node":{"key":"\/apisix\/routes","nodes":{},"dir":true},"count":"1","action":"get"}
```
Then I sent request to Apache APISIX to verify the route by executing
```
kubectl exec -it -n ingress-apisix apisix-6fc55984cb-hwvjq -- curl http://127.0.0.1:9080/headers -H 'Host: local.httpbin.org'
```
I got
```
Defaulted container "apisix" out of: apisix, wait-etcd (init)
{"error_msg":"404 Route Not Found"}
```

I'm not sure what caused `Defaulted container "apisix" out of: apisix, wait-etcd (init)` and the 404 error. Could someone help me figure it out?

---

PS. Here is some context info for debugging purpose.
```
kubectl get po -n=ingress-apisix -o wide
NAME                      READY   STATUS    RESTARTS   AGE   IP           NODE       NOMINATED NODE   READINESS GATES
apisix-6fc55984cb-hwvjq   1/1     Running   0          29m   172.17.0.3   minikube   <none>           <none>
apisix-etcd-0             1/1     Running   0          29m   172.17.0.4   minikube   <none>           <none>


$ kubectl get deploy -n ingress-apisix
NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
apisix               1/1     1            1           34m
ingress-controller   1/1     1            1           46s

$ kubectl get svc -n ingress-apisix
NAME                   TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)             AGE
apisix-admin           NodePort    10.96.3.95      <none>        9180:31552/TCP      35m
apisix-etcd            ClusterIP   10.110.143.17   <none>        2379/TCP,2380/TCP   35m
apisix-etcd-headless   ClusterIP   None            <none>        2379/TCP,2380/TCP   35m
apisix-gateway         NodePort    10.109.35.39    <none>        80:30818/TCP        35m
```

**Comment by tokers** [comment]
@fhuzero Please paste the logs of apisix-ingress-controller when you incur such an error.

**Comment by fhuzero** [comment]
The logs of apisix-ingress-controller is shown below (a bit ugly though).
```
$ kubectl logs --selector=app=apisix-ingress-controller -n ingress-apisix
2021-06-14T16:23:17+08:00	[31merror[0m	apisix/upstream.go:76	failed to get upstream from APISIX	{"name": "ingress-apisix_apisix-gateway_80", "url": "http://127.0.0.1:9080/apisix/admin/upstreams/ffd51960", "cluster": "default", "error": "Get http://127.0.0.1:9080/apisix/admin/upstreams/ffd51960: dial tcp 127.0.0.1:9080: connect: connection refused"}
2021-06-14T16:23:17+08:00	[31merror[0m	ingress/endpoint.go:141	failed to get upstream	{"upstream": "ingress-apisix_apisix-gateway_80", "cluster": "name=default; base_url=http://127.0.0.1:9080/apisix/admin", "error": "Get http://127.0.0.1:9080/apisix/admin/upstreams/ffd51960: dial tcp 127.0.0.1:9080: connect: connection refused"}
2021-06-14T16:23:17+08:00	[33mwarn[0m	ingress/endpoint.go:168	sync endpoints failed, will retry	{"object": {"Type":1,"Object":{"metadata":{"name":"apisix-gateway","namespace":"ingress-apisix","uid":"3a5d055a-3d2d-4547-9949-157b9bca5db3","resourceVersion":"649","creationTimestamp":"2021-06-14T08:10:12Z","labels":{"app.kubernetes.io/instance":"apisix","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"apisix","app.kubernetes.io/version":"2.6.0","helm.sh/chart":"apisix-0.3.4"},"annotations":{"endpoints.kubernetes.io/last-change-trigger-time":"2021-06-14T08:10:13Z"},"managedFields":[{"manager":"kube-controller-manager","operation":"Update","apiVersion":"v1","time":"2021-06-14T08:10:14Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:endpoints.kubernetes.io/last-change-trigger-time":{}},"f:labels":{".":{},"f:app.kubernetes.io/instance":{},"f:app.kubernetes.io/managed-by":{},"f:app.kubernetes.io/name":{},"f:app.kubernetes.io/version":{},"f:helm.sh/chart":{}}},"f:subsets":{}}}]},"subsets":[{"notReadyAddresses":[{"ip":"172.17.0.3","nodeName":"minikube","targetRef":{"kind":"Pod","namespace":"ingress-apisix","name":"apisix-6fc55984cb-ggcwz","uid":"603fc1f8-3275-46f4-9f39-0802378ad6be","resourceVersion":"646"}}],"ports":[{"name":"apisix-gateway","port":9080,"protocol":"TCP"}]}]},"Tombstone":null}}
2021-06-14T16:23:17+08:00	[31merror[0m	apisix/upstream.go:76	failed to get upstream from APISIX	{"name": "ingress-apisix_apisix-admin_9180", "url": "http://127.0.0.1:9080/apisix/admin/upstreams/ff9d87bf", "cluster": "default", "error": "Get http://127.0.0.1:9080/apisix/admin/upstreams/ff9d87bf: dial tcp 127.0.0.1:9080: connect: connection refused"}
2021-06-14T16:23:17+08:00	[31merror[0m	ingress/endpoint.go:141	failed to get upstream	{"upstream": "ingress-apisix_apisix-admin_9180", "cluster": "name=default; base_url=http://127.0.0.1:9080/apisix/admin", "error": "Get http://127.0.0.1:9080/apisix/admin/upstreams/ff9d87bf: dial tcp 127.0.0.1:9080: connect: connection refused"}
2021-06-14T16:23:17+08:00	[33mwarn[0m	ingress/endpoint.go:168	sync endpoints failed, will retry	{"object": {"Type":1,"Object":{"metadata":{"name":"apisix-admin","namespace":"ingress-apisix","uid":"2317c22e-a277-4af9-b3cc-247b0e0e2037","resourceVersion":"647","creationTimestamp":"2021-06-14T08:10:12Z","labels":{"app.kubernetes.io/instance":"apisix","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"apisix","app.kubernetes.io/version":"2.6.0","helm.sh/chart":"apisix-0.3.4"},"annotations":{"endpoints.kubernetes.io/last-change-trigger-time":"2021-06-14T08:10:13Z"},"managedFields":[{"manager":"kube-controller-manager","operation":"Update","apiVersion":"v1","time":"2021-06-14T08:10:14Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:endpoints.kubernetes.io/last-change-trigger-time":{}},"f:labels":{".":{},"f:app.kubernetes.io/instance":{},"f:app.kubernetes.io/managed-by":{},"f:app.kubernetes.io/name":{},"f:app.kubernetes.io/version":{},"f:helm.sh/chart":{}}},"f:subsets":{}}}]},"subsets":[{"notReadyAddresses":[{"ip":"172.17.0.3","nodeName":"minikube","targetRef":{"kind":"Pod","namespace":"ingress-apisix","name":"apisix-6fc55984cb-ggcwz","uid":"603fc1f8-3275-46f4-9f39-0802378ad6be","resourceVersion":"646"}}],"ports":[{"name":"apisix-admin","port":9180,"protocol":"TCP"}]}]},"Tombstone":null}}
2021-06-14T16:23:17+08:00	[31merror[0m	apisix/upstream.go:76	failed to get upstream from APISIX	{"name": "ingress-apisix_apisix-etcd_2380", "url": "http://127.0.0.1:9080/apisix/admin/upstreams/d7f2bf2e", "cluster": "default", "error": "Get http://127.0.0.1:9080/apisix/admin/upstreams/d7f2bf2e: dial tcp 127.0.0.1:9080: connect: connection refused"}
2021-06-14T16:23:17+08:00	[31merror[0m	ingress/endpoint.go:141	failed to get upstream	{"upstream": "ingress-apisix_apisix-etcd_2380", "cluster": "name=default; base_url=http://127.0.0.1:9080/apisix/admin", "error": "Get http://127.0.0.1:9080/apisix/admin/upstreams/d7f2bf2e: dial tcp 127.0.0.1:9080: connect: connection refused"}
2021-06-14T16:23:17+08:00	[33mwarn[0m	ingress/endpoint.go:168	sync endpoints failed, will retry	{"object": {"Type":1,"Object":{"metadata":{"name":"apisix-etcd","namespace":"ingress-apisix","uid":"a6bceedb-4a9e-495e-b652-fd056c419ece","resourceVersion":"670","creationTimestamp":"2021-06-14T08:10:12Z","labels":{"app.kubernetes.io/instance":"apisix","app.kubernetes.io/managed-by":"Helm","app.kubernetes.io/name":"etcd","helm.sh/chart":"etcd-5.2.1"},"annotations":{"endpoints.kubernetes.io/last-change-trigger-time":"2021-06-14T08:10:15Z"},"managedFields":[{"manager":"kube-controller-manager","operation":"Update","apiVersion":"v1","time":"2021-06-14T08:10:22Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:endpoints.kubernetes.io/last-change-trigger-time":{}},"f:labels":{".":{},"f:app.kubernetes.io/instance":{},"f:app.kubernetes.io/managed-by":{},"f:app.kubernetes.io/name":{},"f:helm.sh/chart":{}}},"f:subsets":{}}}]},"subsets":[{"notReadyAddresses":[{"ip":"172.17.0.4","nodeName":"minikube","targetRef":{"kind":"Pod","namespace":"ingress-apisix","name":"apisix-etcd-0","uid":"273da47c-c724-44f9-bb49-a962d21f9319","resourceVersion":"666"}}],"ports":[{"name":"peer","port":2380,"protocol":"TCP"},{"name":"client","port":2379,"protocol":"TCP"}]}]},"Tombstone":null}}
2021-06-14T16:23:17+08:00	[33mwarn[0m	ingress/endpoint.go:93	service kube-system/k8s.io-minikube-hostpath was deleted

```

**Comment by tokers** [comment]
@fhuzero It seems that the APISIX address was not configured correctly as all requests were sent to `127.0.0.1:9080`.

**Comment by fhuzero** [comment]
> @fhuzero It seems that the APISIX address was not configured correctly as all requests were sent to `127.0.0.1:9080`.

I followed instructions in [proxy-the-httpbin-service](https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md) step by step exactly as it says. However, I did not notice instructions on how to configure APISIX address. I'm not sure whether and how one should update `apisix.base_url` field in `samples/deploy/configmapapisix-ingress-cm.yaml`. Could you please give more specific guidance on that?

By the way, I installed apisix-ingress-controller by Kustomize with one change. That is, I updated container image version to the latest release version in `samples/deploy/deployment/ingress-controller.yaml`. I may propose a PR to update the file after this issue being solved.

**Comment by tokers** [comment]
> I followed instructions in proxy-the-httpbin-service step by step exactly as it says. However, I did not notice instructions on how to configure APISIX address. I'm not sure whether and how one should update apisix.base_url field in samples/deploy/configmapapisix-ingress-cm.yaml. Could you please give more specific guidance on that?

We have a lot of installation documents, for instance, you can see [Install Ingress APISIX on ACK
](http://apisix.apache.org/docs/ingress-controller/deployments/ack), there are some guides that you can refer, including the `baseURL` configuration.

> By the way, I installed apisix-ingress-controller by Kustomize with one change. That is, I updated container image version to the latest release version in `samples/deploy/deployment/ingress-controller.yaml`. I may propose a PR to update the file after this issue being solved.

Thanks!

**Comment by fhuzero** [comment]
By installing apisix-ingress-controller via Helm Charts (following [Install Ingress APISIX on Minikube](https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/deployments/minikube.md)), I have got expected response (Hooray!) in this [example](https://github.com/apache/apisix-ingress-controller/blob/master/docs/en/latest/practices/proxy-the-httpbin-service.md#test), except with a `Defaulted container "apisix" out of: apisix, wait-etcd (init)` before the response. I suppose it is fine.

But here is my problem. When I tried to install apisix-ingress-controller by Kustomize according to [this](https://github.com/apache/apisix-ingress-controller/blob/master/install.md#kustomize-support), the instruction says

> Parameters are hardcoded so if the default values are not good for you, just tweak them manually.

So I changed the container image version to the latest one. And also changed `apisix.base_url` to `http://apisix-admin:9180/apisix/admin`, instead of its default `http://127.0.0.1:9080/apisix/admin`, just as `--set config.apisix.baseURL=http://apisix-admin:9180/apisix/admin` when installing apisix-ingress-controller by Helm Charts.  Then I cannot pass the test and got 404 error message like before. IMHO, [Kustomize Support](https://github.com/apache/apisix-ingress-controller/blob/master/install.md#kustomize-support) part is not very easy to understand. Is there any other values need to be _tweaked manully_?

**Comment by fhuzero** [comment]
I think this issue can be closed since it has been solved by #546. Besides, I have opened a new issue #554 for my question about installation via Kustomize, which is off-topic in this issue.

**Comment by tokers** [comment]
> I think this issue can be closed since it has been solved by #546. Besides, I have opened a new issue #554 for my question about installation via Kustomize, which is off-topic in this issue.

I'd like to know whether @bahtou still has problems.

**Comment by gxthrj** [comment]
The issue is stale, feel free to reopen.
